scripts/generate-entry-points.js

```diff
@@ -7,67 +7,43 @@ const projectRoot = path.dirname(currentPath);
+        title: 'dYdX Trading Rewards | Earn $DYDX Tokens Through Trading & Staking',
+        description: 'Maximize your trading on dYdX and earn $DYDX token rewards. Engage with the platform, contribute to the liquidity, and get rewarded.',
         fileName: 'trading-rewards.html',
     },
-    {
-        title: 'Trade ETH-USD on dYdX Chain | Ethereum Trading with Deep Liquidity',
-        description: 'Engage in Ethereum trading on dYdX Chain with ETH-USD pair, benefiting from deep liquidity, tight spreads, and advanced trading features.',
-        fileName: 'trade-ETH-USD.html',
-    },
-    {
-        title: 'Trade BTC-USD on dYdX Chain | Bitcoin Trading with Deep Liquidity',
-        description: 'Engage in Bitcoin trading on dYdX Chain with BTC-USD pair, benefiting from deep liquidity, tight spreads, and advanced trading features.',
-        fileName: 'trade-BTC-USD.html',
-    },
-    {
-        title: 'Trade SOL-USD on dYdX Chain | Solana Trading with Deep Liquidity',
-        description: 'Engage in Solana trading on dYdX Chain with SOL-USD pair, benefiting from deep liquidity, tight spreads, and advanced trading features.',
-        fileName: 'trade-SOL-USD.html',
-    },
-    {
-        title: 'Trade AVAX-USD on dYdX Chain | Avalanche Trading with Deep Liquidity',
-        description: 'Engage in Avalanche trading on dYdX Chain with AVAX-USD pair, benefiting from deep liquidity, tight spreads, and advanced trading features.',
-        fileName: 'trade-AVAX-USD.html',
-    },
-    {
-        title: 'Trade LINK-USD on dYdX Chain | Chainlink Trading with Deep Liquidity',
-        description: 'Engage in Chainlink trading on dYdX Chain with LINK-USD pair, benefiting from deep liquidity, tight spreads, and advanced trading features.',
-        fileName: 'trade-LINK-USD.html',
-    },
-    {
-        title: 'Trade MATIC-USD on dYdX Chain | Polygon Trading with Deep Liquidity',
-        description: 'Engage in Polygon trading on dYdX Chain with MATIC-USD pair, benefiting from deep liquidity, tight spreads, and advanced trading features.',
-        fileName: 'trade-MATIC-USD.html',
-    },
-    {
-        title: 'Trade DOGE-USD on dYdX Chain | Dogecoin Trading with Deep Liquidity',
-        description: 'Engage in Dogecoin trading on dYdX Chain with DOGE-USD pair, benefiting from deep liquidity, tight spreads, and advanced trading features.',
-        fileName: 'trade-DOGE-USD.html',
-    },
 ];
 
 try {
+    console.log(`Fetching top markets from ${TOP_MARKETS_URL} ...`);
+    for (const market of await fetchTopMarkets(TOP_MARKETS_URL)) {
+        ENTRY_POINTS.push({
+            title: `Trade ${market} on dYdX | DeFi's pro decentralized trading platform`,
+            description: `Engage in decentralized ${market} trading on dYdX, benefit from deep liquidity, self-custody, and advanced trading features.`,
+            fileName: `trade-${market}.html`,
+        });
+    }
+
     fs.mkdir(entryPointsDir, { recursive: true });
```

Comment:
I know this wasn't changed by this PR, but should this be await, so that  the directory exists before the the code below?
```suggestion
    await fs.mkdir(entryPointsDir, { recursive: true });
```

Reply:
good catch, adding